File: run_app.py
import subprocess
import sys
import os
import time
import signal

def run_services():
    """
    Запускает React Frontend и FastAPI Backend в параллельных процессах.
    """
    print("==================================================")
    print("   TenderSmart Gidroizol AI - ЗАПУСК СИСТЕМЫ")
    print("==================================================")

    # Определяем команду для npm в зависимости от ОС
    # Windows требует npm.cmd, Linux/Mac просто npm
    npm_cmd = "npm.cmd" if os.name == 'nt' else "npm"
    
    # Пути
    project_root = os.getcwd()
    
    processes = []

    try:
        # 1. Запуск Frontend (Vite)
        print(f"🚀 Запуск Frontend ({npm_cmd} run dev)...")
        frontend_process = subprocess.Popen(
            [npm_cmd, "run", "dev"],
            cwd=project_root,
            shell=False, # Shell=False безопаснее и позволяет лучше управлять процессом
            env=os.environ.copy()
        )
        processes.append(frontend_process)

        # Небольшая пауза, чтобы фронт успел инициализироваться (опционально)
        time.sleep(1)

        # 2. Запуск Backend (Python)
        # Используем текущий интерпретатор python (из venv)
        python_executable = sys.executable 
        print(f"🐍 Запуск Backend ({python_executable} backend.py)...")
        
        backend_process = subprocess.Popen(
            [python_executable, "backend.py"],
            cwd=project_root,
            env=os.environ.copy()
        )
        processes.append(backend_process)

        print("\n✅ Оба сервиса запущены. Нажмите Ctrl+C для остановки.\n")
        
        # Ожидаем завершения процессов (фактически висит, пока не нажмем Ctrl+C)
        backend_process.wait()
        frontend_process.wait()

    except KeyboardInterrupt:
        print("\n\n🛑 Остановка сервисов...")
    finally:
        # Принудительное завершение при выходе
        for p in processes:
            if p.poll() is None: # Если процесс еще жив
                p.terminate()
                # p.kill() # Если terminate не сработает
        print("Система остановлена.")

if __name__ == "__main__":
    run_services()

File: run_app.js
import { spawn } from 'child_process';
import fs from 'fs';
import path from 'path';

console.log('🚀 run_app.js is starting!');

const npmCmd = process.platform === 'win32' ? 'npm.cmd' : 'npm';
console.log(`🚀 Starting Frontend (${npmCmd} run dev:frontend)...`);

const frontend = spawn(npmCmd, ['run', 'dev:frontend'], {
  stdio: 'inherit',
  shell: true
});

const venvPython = '.\\\\.venv\\\\Scripts\\\\python.exe';
const venvPythonPath = path.join(process.cwd(), '.venv', 'Scripts', 'python.exe');

if (!fs.existsSync(venvPythonPath)) {
  console.error('Не найден Python виртуального окружения .venv. Сначала создайте/установите зависимости в виртуальное окружение.');
  process.exit(1);
}

console.log(`🚀 Starting Backend (${venvPython} run_backend.py)...`);

const backend = spawn(venvPython, ['run_backend.py'], {
  stdio: 'inherit',
  shell: true
});

process.on('SIGINT', () => {
  frontend.kill('SIGINT');
  backend.kill('SIGINT');
  process.exit();
});
